File: client/java/src/main/java/com/be/mesh/client/prsim/RuntimeHook.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.prsim;

/**
 * @author coyzeng@gmail.com
 */
public interface RuntimeHook {

    /**
     * Trigger when mesh runtime is start.
     */
    default void start() throws Throwable {

    }

    /**
     * Trigger when mesh runtime is stop.
     */
    default void stop() throws Throwable {

    }

    /**
     * Trigger then mesh runtime context is refresh or metadata is refresh.
     */
    default void refresh() throws Throwable {

    }

}

File: client/python/mesh/prsim/runtime_hook.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from abc import abstractmethod, ABC

from mesh.macro import spi


@spi("mesh")
class RuntimeHook(ABC):

    @abstractmethod
    def start(self):
        """
         Trigger when mesh runtime is start.
        :return:
        """
        pass

    @abstractmethod
    def stop(self):
        """
        Trigger when mesh runtime is stop.
        :return:
        """
        pass

    @abstractmethod
    def refresh(self):
        """
        Trigger then mesh runtime context is refresh or metadata is refresh.
        :return:
        """
        pass
